src/extension/Daemon.js

```diff
@@ -346,6 +346,16 @@ var Daemon = class Daemon {
 
     return 0;
   }
+  
+  // Opens a menu with that name if there are currently none open. Closes the currently 
+  // open menu otherwise.   
+  ToggleMenu(name) {
+    if (this._menu.getID() == null) {
+      return ShowMenu(name);
+    } else {
+      return CancelMenu();
+    }
```

Comment:
Here are two `this` missing:

```suggestion
    if (this._menu.getID() == null) {
      return this.ShowMenu(name);
    } else {
      return this.CancelMenu();
    }
```